src/main/java/org/terasology/metalrenegades/ai/system/NeedsSystem.java

```diff
@@ -18,35 +18,94 @@
 import org.terasology.entitySystem.entity.EntityManager;
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
-import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
 import org.terasology.metalrenegades.ai.CitizenNeed;
 import org.terasology.metalrenegades.ai.component.NeedsComponent;
+import org.terasology.metalrenegades.ai.component.NeedsSystemConfigurationComponent;
+import org.terasology.metalrenegades.ai.event.CitizenSpawnedEvent;
 import org.terasology.registry.In;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
 import org.terasology.world.time.WorldTimeEvent;
 
+import java.util.Iterator;
+import java.util.Optional;
+
 /**
  * Manages needs for all citizens with {@link NeedsComponent}.
  */
 @RegisterSystem(value = RegisterMode.AUTHORITY)
 public class NeedsSystem extends BaseComponentSystem {
 
+    @In
+    private PrefabManager prefabManager;
+
     @In
     private EntityManager entityManager;
 
+    private NeedsSystemConfigurationComponent configComponent;
+
+    private Random random;
+
+    @Override
+    public void initialise() {
+        random = new FastRandom();
+
+        for (Prefab prefab : prefabManager.listPrefabs(NeedsSystemConfigurationComponent.class)) {
+            NeedsSystemConfigurationComponent newConfig = prefab.getComponent(NeedsSystemConfigurationComponent.class);
+
+            if (configComponent == null || configComponent.priority < newConfig.priority) {
+                configComponent = newConfig;
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onCitizenSpawned(CitizenSpawnedEvent citizenSpawnedEvent, EntityRef citizen) {
+        NeedsComponent needsComponent = new NeedsComponent();
+        configComponent.needsConfigs.stream().forEach(config -> {
```

Comment:
[Lists](https://docs.oracle.com/javase/8/docs/api/java/util/List.html) (iterables) don't need a stream and can directly [`forEach`](https://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html#forEach-java.util.function.Consumer-)